scripts/tb.sh

```diff
@@ -0,0 +1,290 @@
+                ;;
+            *)
+                positional_args+=("$1")
+                ;;
+        esac
+        shift
+    done
+
+    set -- "${positional_args[@]}"
+
+    if [ $# -lt 1 ]; then
+        usage_error "Wrong number of positional arguments"
+    fi
+    ACTION="$1"
+
+    case $ACTION in
+        modify|reset|build)
+            if [ $# -ne 2 ]; then
+                usage_error "Wrong number of arguments"
+            fi
+            RECIPE_ARG="$2"
+            ;;
+        deploy)
+            if [ $# -ne 3 ]; then
+                usage_error "Wrong number of positional arguments"
+            fi
+            RECIPE_ARG="$2"
+            DESTINATION_ARG="$3"
+            if [ "$DESTINATION_ARG" == "/" ]; then
+                read -p "Destination path is '/', are you sure? [N|y]: " -n 1 -r choice
+                echo
+                case $choice in
+                    y) ;;
+                    *) exit 0
+                esac
+            fi
+            ;;
+        menuconfig)
+            RECIPE_ARG="linux-tb"
+            ;;
+        *)
+            usage_error "Wrong positional argument"
+            ;;
+    esac
+
+    case $RECIPE_ARG in
+        skl|grub|grub-efi|linux-tb|tb-minimal-image)
+            ;;
+        *)
+            usage_error "Wrong recipe"
+            ;;
+    esac
+}
+
+menuconfig() {
+    if [ -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        kas-container shell "$KAS_YAML" \
+            -c "devtool menuconfig $RECIPE_ARG"
+    else
+        error_msg "First use ./tb.sh modify linux-tb"
+    fi
+}
+
+modify_recipe() {
+    if [ ! -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        kas-container shell "$KAS_YAML" \
+            -c "devtool modify $RECIPE_ARG"
+    else
+        error_msg "Source is already fetched"
+    fi
+}
+
+reset_recipe() {
+    if [ -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        kas-container shell "$KAS_YAML" \
+            -c "devtool reset $RECIPE_ARG"
+    else
+        error_msg "Nothing to reset"
+    fi
+}
+
+build_recipe() {
+    if [ "$RECIPE_ARG" == "tb-minimal-image" ]; then
+        rm -rf "build/workspace/sources/grub/gnulib" 2&>/dev/null|| true
+        rm -rf "build/workspace/sources/grub-efi/gnulib" 2&>/dev/null || true
+        kas-container shell "$KAS_YAML" \
+            -c "devtool build-image $RECIPE_ARG"
+        return
+    fi
+    if [ ! -d "build/workspace/sources/$RECIPE_ARG" ]; then
+        modify_recipe
+    elif [ "$RECIPE_ARG" == "grub" ] || [ "$RECIPE_ARG" == "grub-efi" ]; then
+        rm -rf "build/workspace/sources/$RECIPE_ARG/gnulib"
+    fi
+
+    kas-container shell "$KAS_YAML" \
+        -c "devtool build $RECIPE_ARG"
+}
+
+deploy_recipe() {
+    local work_dir="build/tmp/work"
+    local deploy_dir="build/tmp/deploy/images/genericx86-64"
+    local core_path="$work_dir/core2-64-tb-linux"
+    local genericx86_path="$work_dir/genericx86_64-tb-linux"
+    local kernel_path="$genericx86_path/linux-tb/6.6.1"
```

Comment:
Is it possible to unhardcode version number? This place is likely to be missed on next upgrade.

Reply:
fixed in [3ba2257](https://github.com/3mdeb/meta-trenchboot/pull/29/commits/3ba2257a93bfbfc634d83f4f3c4f5bd194fc63c0)